### PR TITLE
feat: Add usertype slector and action, show user type(role) on user i…

### PR DESCRIPTION
**Pull request by aleciafb**
Chores
- removed unused get Token selector

Actions:
- Fetch all user type
- Fetch single user type
- Fetch usertypes when fetching users
- Fetch usertype when fetching single user
- Fetch user types when user redirects to usertypes page

Templates:
- Display list of users
- Add userType navigation

Refactor:
- Display field component

Utils:
- Add a helper function to get resource id from URL (TO BE REFACTORED!)
- Format date helper

Notes:
In the future, I need to refactor the index views to render table using a generic component - just want to add pagination on one of these tables first.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
   "dependencies": {
     "@material-ui/core": "^1.1.0",
     "@material-ui/icons": "^1.1.0",
+    "dateformat": "^3.0.3",
```
Comment (jwfwessels) [inline package.json]: do you really need a lib for this? the built in date lib is pretty powerful?

File: src/actions/userTypes.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+export const fetch = () => async dispatch => {
+    dispatch(fetchList('userTypes')());
+    const users = await dispatch(api.getUserTypes());
+    return await dispatch(receiveList('userTypes')(users));
+};
+
+export const fetchById = id => async dispatch => {
+    dispatch(fetchItem('userTypes')());
+    const userType = await dispatch(api.getUserTypeById(id));
+    return await dispatch(reciveItem('userTypes')(userType));
```
Comment (jwfwessels) [inline src/actions/userTypes.js]: typo `receive`, not `recive`

File: src/actions/users.js [hunk 3]
```diff
@@ -4,17 +4,24 @@ import {
...
     return await dispatch(receiveList('users')(users));
 };
 
-export const fetchById = id => async (dispatch, getState, subscribe) => {
+export const fetchById = id => async dispatch => {
     dispatch(fetchItem('users')());
     const user = await dispatch(api.getUserById(id));
+    const userTypeId = getResourceIdFromUrl(user.user_type);
+    await dispatch(fetchUserTypeItem(userTypeId));
     return await dispatch(reciveItem('users')(user));
```
Comment (jwfwessels) [inline src/actions/users.js]: same typo ^

File: src/containers/User.js [hunk 4]
```diff
@@ -1,15 +1,30 @@
...
+import { getResourceIdFromUrl } from './../utils';
 
-import { getUserById, isUserLoading } from '../reducers';
+import { getUserById, getUserTypeById, isUserLoading } from '../reducers';
+
+const mapStateToProps = (
+    state,
+    {
+        match: {
+            params: { id }
```
Comment (jwfwessels) [inline src/containers/User.js]: will id always exist? else this will throw an error. (should be fine just asking)

File: src/containers/User.js [hunk 5]
```diff
@@ -1,15 +1,30 @@
...
+        }
+    }
+) => {
+    const user = getUserById(state, id);
+    let userType;
+
+    user &&
```
Comment (jwfwessels) [inline src/containers/User.js]: in what situation would user be undefined?

File: src/utils.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import dateFormat from 'dateformat';
+
+export const getResourceIdFromUrl = url => {
+    // TODO We need a better way to get id's from URLS
+    // Current assumption is that it is always the last paramater
+    return url.split('/').slice(-2)[0];
+};
+
+export const formatDateTime = dateTimeString => {
+    return dateFormat(new Date(dateTimeString), 'dd/mm/yyyy, h:MM TT');
```
Comment (jwfwessels) [inline src/utils.js]: what does dateFormat do? just return a string cause the Date object has str methods with formaters 😅

File: src/actions/userTypes.js (hunk 2 above)
Comment (aleciafb) [reply src/actions/userTypes.js]: Thanks!!

File: src/containers/User.js (hunk 4 above)
Comment (aleciafb) [reply src/containers/User.js]: In this case this component always renders one user thus an id is needed. I need to add a error boundary and handle these sort of things (Coming in a PR or 2)

File: src/containers/User.js (hunk 5 above)
Comment (aleciafb) [reply src/containers/User.js]: If it's still loading

File: src/utils.js (hunk 6 above)
Comment (aleciafb) [reply src/utils.js]: @jwfwessels `toLocaleFormat` is depricated. `toLocaleDateString` and `toLocaleTime` formatting options are not supported on Firefox. Used lightweight library in this case

File: package.json (hunk 1 above)
Comment (aleciafb) [reply package.json]: See last comment and DMd you my reasoning behind this :)

**Pull request merged**